src/wallet/wallet.cpp

```diff
@@ -4286,6 +4328,223 @@ bool CWallet::CreateZerocoinMintModelV2(string &stringError, const string& denom
+
+                if ((mintHeight = zerocoinState->GetMintedCoinHeightAndId(coinPublicValue, (int)d, mintId)) > 0 &&
+                        IsZerocoinTxV2(d, Params().GetConsensus(), mintId) == (coinVersion == ZEROCOIN_TX_VERSION_2)) {
+                    if (mintEntry) {
+                        *mintEntry = coin;
+
+                        // we don't trust values in DB, we use values from blockchain
+                        mintEntry->nHeight = mintHeight;
+                        mintEntry->id = mintId;
+
+                        // no need to find anything else
+                        return 1;
+                    }
+                    else {
+                        result++;
+                    }
+                }
+            }
+        }
+    }
+
+    return result;
+}
+
+bool CWallet::CreateZerocoinToSigmaRemintModel(string &stringError, int version, libzerocoin::CoinDenomination denomination, CWalletTx *wtx) {
+    // currently we don't support zerocoin mints v1
+    assert(version == ZEROCOIN_TX_VERSION_2);
+
+    if (IsLocked()) {
+        stringError = "Error: Wallet locked, unable to create transaction!";
+        return false;
+    }
+
+    CWalletTx wtxNew;
+    wtxNew.BindWallet(this);
+    CMutableTransaction txNew;
+
+    LOCK2(cs_main, cs_wallet);
+
+    txNew.vin.clear();
+    txNew.vout.clear();
+    txNew.wit.SetNull();
+    wtxNew.fFromMe = true;
+
+    CZerocoinEntry mintEntry;
+    if (GetNumberOfUnspentMintsForDenomination(version, denomination, &mintEntry) <= 0 || mintEntry.nHeight > chainActive.Height()) {
+        stringError = "No zerocoin mints found";
+        return false;
+    }
+
+    // Create remint tx input without signature (metadata is unknown at the moment)
+    sigma::CoinRemintToV3 remint(version, (unsigned)denomination, (unsigned)mintEntry.id,
+            mintEntry.serialNumber, mintEntry.randomness, chainActive[mintEntry.nHeight]->GetBlockHash(),
+            mintEntry.ecdsaSecretKey);
+
+    CDataStream ds1(SER_NETWORK, PROTOCOL_VERSION);
+    ds1 << remint;
+
+    CScript remintScriptBeforeSignature;
+    remintScriptBeforeSignature << OP_ZEROCOINTOSIGMAREMINT;
+    remintScriptBeforeSignature.insert(remintScriptBeforeSignature.end(), ds1.begin(), ds1.end());
+
+    CTxIn remintTxIn;
+    remintTxIn.nSequence = mintEntry.id;
+    remintTxIn.scriptSig = remintScriptBeforeSignature;
+    remintTxIn.prevout.SetNull();
+    txNew.vin.push_back(remintTxIn);
+
+    vector<sigma::PrivateCoin> privateCoins;
+
+    static struct ZerocoinToSigmaMintDenominationMap {
+        int zerocoinDenomination;
+        sigma::CoinDenomination sigmaDenomination;
+        int numberOfSigmaMints;
+    } zerocoinToSigmaDenominationMap[] = {
+        {1, sigma::CoinDenomination::SIGMA_DENOM_1, 1},
+        {10, sigma::CoinDenomination::SIGMA_DENOM_10, 1},
+        {25, sigma::CoinDenomination::SIGMA_DENOM_25, 1},
+        {50, sigma::CoinDenomination::SIGMA_DENOM_10, 5},
```

Comment:
Why not 2x25?

Reply:
Good catch! It's a leftover from before 25 was introduced. Will change